contracts/bridges/WormholeMessenger.sol

```diff
@@ -0,0 +1,209 @@
+error ZeroValue();
+
+/// @dev Only self contract is allowed to call the function.
+/// @param sender Sender address.
+/// @param instance Required contract instance address.
+error SelfCallOnly(address sender, address instance);
+
+/// @dev Only `wormholeRelayer` is allowed to call the function.
+/// @param sender Sender address.
+/// @param wormholeRelayer Required L2 Wormhole Relayer address.
+error wormholeRelayerOnly(address sender, address wormholeRelayer);
+
+/// @dev Wrong source chain Id.
+/// @param received Chain Id received.
+/// @param required Required chain Id.
+error WrongSourceChainId(uint256 received, uint256 required);
+
+/// @dev Only on behalf of `sourceGovernor` the function is allowed to process the data.
+/// @param sender Sender address.
+/// @param sourceGovernor Required Source Governor address.
+error SourceGovernorOnly(address sender, address sourceGovernor);
+
+/// @dev The message with a specified hash has already been delivered.
+/// @param deliveryHash Delivery hash.
+error AlreadyDelivered(bytes32 deliveryHash);
+
+/// @dev Provided incorrect data length.
+/// @param expected Expected minimum data length.
+/// @param provided Provided data length.
+error IncorrectDataLength(uint256 expected, uint256 provided);
+
+/// @dev Provided value is bigger than the actual balance.
+/// @param value Provided value.
+/// @param balance Actual balance.
+error InsufficientBalance(uint256 value, uint256 balance);
+
+/// @dev Target execution failed.
+/// @param target Target address.
+/// @param value Provided value.
+/// @param payload Provided payload.
+error TargetExecFailed(address target, uint256 value, bytes payload);
+
+/// @title WormholeMessenger - Smart contract for the governor bridge communication via wormhole
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+/// @author Mariapia Moscatiello - <mariapia.moscatiello@valory.xyz>
+contract WormholeMessenger {
+    event FundsReceived(address indexed sender, uint256 value);
+    event SourceGovernorUpdated(address indexed sourceMessageSender);
+    event MessageReceived(address indexed sourceMessageSender, bytes data, bytes32 deliveryHash, uint256 sourceChain);
+
+    // Default payload data length includes the number of bytes of at least one address (20 bytes or 160 bits),
+    // value (12 bytes or 96 bits) and the payload size (4 bytes or 32 bits)
+    uint256 public constant DEFAULT_DATA_LENGTH = 36;
+    // L2 Wormhole Relayer address that receives the message across the bridge from the source L1 network
+    address public immutable wormholeRelayer;
+    // Source governor chain Id
+    uint16 public immutable sourceGovernorChainId;
+    // Source governor address on L1 that is authorized to propagate the transaction execution across the bridge
+    address public sourceGovernor;
+    // Mapping of delivery hashes
+    mapping(bytes32 => bool) public mapDeliveryHashes;
```

Comment:
These are two major differences for the wormhole contract - we need to check the source chain Id, and the delivery hash such that it does not repeat.